tools/connector-dev-server/api.go

```diff
@@ -0,0 +1,562 @@
+		if id == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "connectorId is required"})
+			return
+		}
+		pluginsMu.RLock()
+		e, ok := pluginsByID[id]
+		pluginsMu.RUnlock()
+		if !ok {
+			w.WriteHeader(http.StatusNotFound)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "plugin not found"})
+			return
+		}
+		_ = json.NewEncoder(w).Encode(map[string]any{
+			"connectorId": e.ConnectorID.String(),
+			"provider":    e.Provider,
+			"name":        e.Name,
+		})
+	})
+
+	r.Get("/connectors/{connectorId}/state", func(w http.ResponseWriter, req *http.Request) {
+		id := chi.URLParam(req, "connectorId")
+		if id == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "connectorId is required"})
+			return
+		}
+		pluginsMu.RLock()
+		e, ok := pluginsByID[id]
+		pluginsMu.RUnlock()
+		if !ok {
+			w.WriteHeader(http.StatusNotFound)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "plugin not found"})
+			return
+		}
+		_ = json.NewEncoder(w).Encode(map[string]any{
+			"fetch": map[string]cachedState{
+				"accounts":          e.StateCache["fetch/accounts"],
+				"external-accounts": e.StateCache["fetch/external-accounts"],
+				"balances":          e.StateCache["fetch/balances"],
+				"payments":          e.StateCache["fetch/payments"],
+			},
+		})
+	})
+
+	r.Post("/fetch/accounts", func(w http.ResponseWriter, req *http.Request) {
+		// Only works when exactly one plugin is instantiated
+		pluginsMu.RLock()
+		var p models.Plugin
+		if len(pluginsByID) == 1 {
+			for _, e := range pluginsByID {
+				p = e.Plugin
+			}
+		}
+		pluginsMu.RUnlock()
+
+		if p == nil {
+			w.WriteHeader(http.StatusConflict)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "requires exactly one plugin; use /connectors/{connectorId}/fetch/accounts"})
+			return
+		}
+
+		var body models.FetchNextAccountsRequest
+		if err := json.NewDecoder(req.Body).Decode(&body); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "invalid JSON body", "details": err.Error()})
+			return
+		}
+
+		resp, err := p.FetchNextAccounts(req.Context(), body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "fetch accounts failed", "details": err.Error()})
+			return
+		}
+
+		_ = json.NewEncoder(w).Encode(resp)
+	})
+
+	r.Post("/fetch/balances", func(w http.ResponseWriter, req *http.Request) {
+		// Only works when exactly one plugin is instantiated
+		pluginsMu.RLock()
+		var p models.Plugin
+		if len(pluginsByID) == 1 {
+			for _, e := range pluginsByID {
+				p = e.Plugin
+			}
+		}
+		pluginsMu.RUnlock()
+
+		if p == nil {
+			w.WriteHeader(http.StatusConflict)
+			_ = json.NewEncoder(w).Encode(map[string]any{"message": "requires exactly one plugin; use /connectors/{connectorId}/fetch/balances"})
+			return
+		}
```

Comment:
maybe move that in a separate method as it's repeted twice (+ we'll want the same for payment & transactions I assume)